### PR TITLE
JAMES-3755 Healthcheck for Rspamd

### PR DESCRIPTION
**Pull request by quantranhong1999**
resolve https://github.com/linagora/james-project/issues/4807

### CODE REVIEW

**Review by Arsnael — APPROVED** [review]

File: third-party/rspamd/src/test/java/org/apache/james/rspamd/DockerClamAV.java [hunk 1]
```diff
@@ -41,7 +42,8 @@ public DockerClamAV(Network network) {
             .withNetwork(network)
             .withCreateContainerCmdModifier(createContainerCmd -> createContainerCmd.withName("james-clamav-test-" + UUID.randomUUID()))
             .withNetworkAliases("clamav")
-            .waitingFor(Wait.forHealthcheck());
+            .waitingFor(Wait.forHealthcheck()
+                .withStartupTimeout(Duration.ofMinutes(2)));
```
Comment (vttranlina) [inline third-party/rspamd/src/test/java/org/apache/james/rspamd/DockerClamAV.java:46]: what do you think if we have a file configured somewhere, or read it from env?

Comment (vttranlina) [reply third-party/rspamd/src/test/java/org/apache/james/rspamd/DockerClamAV.java:46]: of course, it is not for only ClamAV

File: third-party/rspamd/src/main/java/org/apache/james/rspamd/client/RspamdHttpClient.java [hunk 2]
```diff
@@ -104,9 +104,10 @@ public final int hashCode() {
     public static final String CHECK_V2_ENDPOINT = "/checkV2";
     public static final String LEARN_SPAM_ENDPOINT = "/learnspam";
     public static final String LEARN_HAM_ENDPOINT = "/learnham";
+    public static final String PING_ENDPOINT = "/ping";
     private static final int OK = 200;
     private static final int NO_CONTENT = 204;
-    private static final int FORBIDDEN = 403;
+    private static final int UNAUTHORIZED = 401;
```
[the remark this answers is not in the document]
Comment (vttranlina) [reply third-party/rspamd/src/main/java/org/apache/james/rspamd/client/RspamdHttpClient.java:110]: I guess Quan doesn't want to import one more maven dependency for that

**Review by chibenwa — APPROVED** [review]

File: third-party/rspamd/src/test/java/org/apache/james/rspamd/DockerClamAV.java (hunk 1 above)
Comment (quantranhong1999) [reply third-party/rspamd/src/test/java/org/apache/james/rspamd/DockerClamAV.java:46]: I think a timeout env variable may help for different (slow/fast) environments.
But as long as we hardcode that timeout high enough, it is still fine for whatsoever environment. (fast env still finishes starting container regardless of the timeout)
Let's keep that in mind and let that out of this PR scope IMO.

**Review by vttranlina — APPROVED** [review]

**Pull request merged**